### ISSUE TITLE
`error.InvalidHandle` has confusing/wrong usage in places

### ISSUE THREAD

**Issue by squeek502** [ziglang/zig#19084]
### Zig Version

master

### Steps to Reproduce and Observed Behavior

In places, `error.InvalidHandle` is documented to make it seem like it's WASI-specific:

https://github.com/ziglang/zig/blob/aa39e98d9024ab68246ae45b2d67c7eda46f28ab/lib/std/os.zig#L1576-L1577

and in usage its treated as WASI-specific:

https://github.com/ziglang/zig/blob/aa39e98d9024ab68246ae45b2d67c7eda46f28ab/lib/std/child_process.zig#L522

In this particular case, though, `InvalidHandle` is not even a possible error during `open`/`openat` on any platform AFAICT (not returned from `open`/`openZ`/`openat`/`openatWasi`/`openatZ`)

---

In other places, `error.InvalidHandle` is documented with the same WASI doc comment, but the function is not supported on WASI (unless libc is linked I guess? What does wasi-libc do for `realpath`? In any case, `InvalidHandle` isn't a possible return of `realpath` on WASI).

This is in `RealPathError`:

https://github.com/ziglang/zig/blob/aa39e98d9024ab68246ae45b2d67c7eda46f28ab/lib/std/os.zig#L5546-L5547

In reality, `InvalidHandle` is FreeBSD specific in `RealPathError`:

https://github.com/ziglang/zig/blob/aa39e98d9024ab68246ae45b2d67c7eda46f28ab/lib/std/os.zig#L5800

---

Basically, `error.InvalidHandle` needs a general lookover to make sure everything makes sense, because right now it doesn't.

### Expected Behavior

Consistent and accurate documentation/handling of `error.InvalidHandle`
